test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/IntTest.scala

```diff
@@ -0,0 +1,184 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.compiler
+
+import org.junit.Test
+import org.junit.Assert._
+
+/* General note on the way these tests are written:
+ * We leverage the constant folding applied by the Scala compiler to write
+ * sound tests. We always perform the same operation, on the same operands,
+ * once in a way constant folding understands, and once in a way it doesn't.
+ * Since constant folding is performed on the JVM, we know it has the right
+ * semantics.
+ */
+
+class IntTest {
+  import IntTest._
+
+  @Test def `should_support_unary_-`(): Unit = {
```

Comment:
In general, with JUnit, I would suggest we use the method under test as the test method name. At least for tests that target one identifiable method. So here I would suggest `def unary_-()`. It would also reduce the need for backquotes.